app/main/views/users.py

```diff
@@ -57,11 +57,11 @@ def auth_user():
         db.session.commit()
 
         if user.role == 'supplier':
+            vr = supplier_business.get_supplier_messages(user.supplier_code)
             validation_result = (
-                supplier_business
-                .get_supplier_validation_result(user.supplier_code)
-                ._asdict()
+                vr._asdict() if vr else None
             )
+
         else:
             validation_result = None
```

Comment:
Can this `else` block be removed if we shift `validation_result = None` before the `if`?